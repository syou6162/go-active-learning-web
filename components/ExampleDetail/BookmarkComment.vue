<template>
  <b-list-group-item
    class="hatena-bookmark-item"
  >
    <img
      v-lazy="'https://cdn.profile-image.st-hatena.com/users/' + bookmark.user+ '/profile.png'"
      class="hatena-bookmark-comment-user-icon-img"
    >
    <div class="hatena-bookmark-user-link-and-comment-container">
      <a 
        :href="'http://b.hatena.ne.jp/' + bookmark.user"
        target="_blank"
        rel="noopener"
        class="hatena-bookmark-user-link"
      >
        id:{{ bookmark.user }}
      </a>
      <span class="hatena-bookmark-comment">
        {{ bookmark.comment }}
      </span>
      <span class="hatena-bookmark-timestamp">
        {{ bookmark.timestamp }}
      </span>
    </div>
  </b-list-group-item>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import Bookmark from '~/models/Bookmark'

@Component
export default class BookmarkComment extends Vue {
  @Prop()
  bookmark!: Bookmark
}
</script>

<style scoped>
.hatena-bookmark-item {
  padding: 10px 10px;
}
.hatena-bookmark-comment-user-icon-img {
  float: left;
  width: 32px;
  height: 32px;
  margin: 0 10px 0 0;
}
.hatena-bookmark-user-link-and-comment-container {
  overflow: hidden;
}
.hatena-bookmark-user-link {
  margin: 0 0 0 0;
}
.hatena-bookmark-comment {
  margin: 0 4px 0 0;
}
.hatena-bookmark-timestamp {
  display: block;
  color: #999;
  float: right;
}
</style>