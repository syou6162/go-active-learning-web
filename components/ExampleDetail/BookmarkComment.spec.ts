import { createLocalVue, shallowMount } from '@vue/test-utils'
import { BListGroupItem } from 'bootstrap-vue'
import VueLazyLoad from 'vue-lazyload'
import BookmarkComment from '~/components/ExampleDetail/BookmarkComment.vue'

const localVue = createLocalVue();
localVue.use(VueLazyLoad);
localVue.component('b-list-group-item', BListGroupItem)

const bookmark = {
  "comment": "ブックマークのテストです",
  "user": "syou6162",
  "timestamp": "2020/08/26 14:29",
  "tags": ["tag1", "tag2"]
}

describe('BookmarkComment', () => {
  test('個別事例ページの中のはてなブックマークコメントが正しく表示できる', () => {
    const wrapper = shallowMount(BookmarkComment, {
      localVue,
      propsData: {
        bookmark: bookmark,
      }
    })

    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.find('.hatena-bookmark-item').exists()).toBe(true)
    expect(wrapper.find('.hatena-bookmark-user-link').text()).toBe(`id:${bookmark.user}`)
    expect(wrapper.find('.hatena-bookmark-comment').text()).toBe(`${bookmark.comment}`)

    expect(wrapper.element).toMatchSnapshot()
  })
})