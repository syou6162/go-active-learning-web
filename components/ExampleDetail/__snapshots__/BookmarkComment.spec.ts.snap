// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkComment 個別事例ページの中のはてなブックマークコメントが正しく表示できる 1`] = `
<b-list-group-item-stub
  class="hatena-bookmark-item"
  tag="div"
  target="_self"
>
  <img
    class="hatena-bookmark-comment-user-icon-img"
  />
   
  <div
    class="hatena-bookmark-user-link-and-comment-container"
  >
    <a
      class="hatena-bookmark-user-link"
      href="http://b.hatena.ne.jp/syou6162"
      rel="noopener"
      target="_blank"
    >
      
      id:syou6162
    
    </a>
     
    <span
      class="hatena-bookmark-comment"
    >
      
      ブックマークのテストです
    
    </span>
     
    <span
      class="hatena-bookmark-timestamp"
    >
      
      2020/08/26 14:29
    
    </span>
  </div>
</b-list-group-item-stub>
`;
